### DISCUSSION TITLE
Why does turning off manual control via service take effect almost instantly but turning on manual control takes 5-7 seconds to be reflected in the switch attributes?

### DISCUSSION THREAD

**Discussion by ncd7** [basnijholt/adaptive-lighting#956] [General]
Has anyone else noticed this?
If I make a service call to turn off manual control, the light is almost instantly removed from the "manual_control" attribute of the AL switch. 
But if I make a service call to turn on manual control, it takes 5+ seconds for the light to be added to the "manual_control" attribute of the switch.
Does anyone know why this is the case? Also, does the addon internally reflect this immediately or it truly takes this long to take over manual control? 

The reason I've noticed this is I'm using this in a dashboard to allow people to explicitly enable/disable manual control and "user" are complaining (hrmph :P) that turning on manual control doesn't work because they are too impatient to wait 5-6 seconds. And by users I mean the family hehe.

**Comment by gravityrebel** [comment]
Just curious, why are you manually marking it as manual? The integration should be monitoring the lights and will mark it automatically as soon as you make a change.
> Adaptive Lighting is designed to automatically detect when you or another source (e.g., automation) manually changes light settings 🕹️. When this occurs, the affected light is marked as "manually controlled," and Adaptive Lighting will not make further adjustments until the light is turned off and back on or reset using the adaptive_lighting.set_manual_control service call. This feature is available when take_over_control is enabled.

Just make the change you want without worrying about marking it as manually controlled.

**Reply by ncd7** [reply]
Most times, I do not. I simply issue a command to make a change to the lights and let it do its thing.
But because occasionally this fails, I've added an indicator on my dashboard to show if manual control is on or not. It's a very useful indicator also to know visually what the integration thinks (double tapping the little icon switches back to automatic control).
It's this little icon that first made it obvious to me that when manual control is turned OFF, that is acknowledged immediately in the attributes of the sensor, but when it is turned ON it takes a while. 

Separately, I sometimes need to manually turn on.. manual control (via a service call) as part of my automations because sadly things like say activating a hue scene, doesn't always reliably turn on manual control. So I've found it way more robust to first turn on manual control myself and then activate a scene (when it's part of an automation). So this use cases, together with the dashboard above made me wonder why it works this tway.

**Comment by wthueb** [comment]
Old one, still happens though. Doesn't just happen when you call `adaptive_lighting.set_manual_control`, also happens when you activate a new scene and it sets manual control automatically. From what I can tell, `AdaptiveSwitch.set_manual_control_attributes`/`AdaptiveSwitch.fire_manual_control_event` should probably call `async_write_ha_state` to force HA to update the state instead of relying on a different state change/HA to update it automatically. I'm mostly out of my depth here with HA internals though, I'm just guessing.

Since the manual control event is fired consistently and quickly though, the workaround I'm using is
```yaml
automation:
  - id: refresh_adaptive_lighting_manual_control
    alias: Refresh Adaptive Lighting Manual Control
    triggers:
      - trigger: event
        event_type: adaptive_lighting.manual_control
        event_data:
          switch: switch.adaptive_lighting_default
    actions:
      - action: homeassistant.update_entity
        data:
          entity_id:
            - switch.adaptive_lighting_default
    mode: queued
```
which basically does the same thing without modifying the integration code. Works well for me.

FWIW my use case is [here](https://github.com/wthueb/home-assistant-cfg/blob/9023959b54ab2eb599e15903642dbcbfbc1723ba/packages/lights.yaml#L122). I have multi-gang switches, where each switch is a toggle (activating a scene/normal adaptive lighting if it isn't active, otherwise turning off the light). If you go from adaptive -> scene -> adaptive, that last adaptive press won't get the `manual_control` state correct and will turn off the light (without the above automation).
